modules/foreman_compute_profile.py

```diff
@@ -82,38 +165,76 @@
     'name': 'name',
 }
 
+compute_attribute_name_map = {
+    'compute_resource': 'compute_resource',
+    'vm_attrs': 'vm_attrs',
+}
 
-def main(module):
-
-    (module_params, state) = module.parse_params()
-    name = module_params.get('name')
-    updated_name = module_params.get('updated_name')
-
-    module.connect()
-
-    data = {
-        'name': name
-    }
 
-    compute_profile = cement.find_compute_profile(module, name=name, failsafe=True)
+def compute_attribute(module, compute_profile, attributes):
+    # If we don't work on a copy, we hit a TypeError: Value of unknown type: <class 'nailgun.entities.ComputeProfile'> exception
+    # from module.exit_json(changed=changed) method
+    compute_attribute_dict = attributes.copy()
+    for key in ['compute_resource', 'vm_attrs']:
+        if key not in compute_attribute_dict:
+            module.fail_json(msg='compute_attribute must have %s.' % key)
 
-    if state == 'present' and updated_name:
-        data['name'] = updated_name
+    try:
+        compute_attribute = find_compute_attribute(module, compute_profile_name=compute_profile.name,
+                                                   compute_resource_name=compute_attribute_dict['compute_resource'], failsafe=True)
+        compute_attribute_dict['compute_resource'] = find_compute_resource(module, name=compute_attribute_dict['compute_resource'], failsafe=False)
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
 
-    data = cement.sanitize_entity_dict(data, name_map)
+    if 'compute_profile' not in compute_attribute_dict:
+        compute_attribute_dict['compute_profile'] = compute_profile
 
-    changed = cement.naildown_entity_state(nailgun.entities.ComputeProfile, data, compute_profile, state, module)
+    changed = naildown_entity_state(ComputeAttribute, compute_attribute_dict, compute_attribute, 'present', module)
 
     return changed
 
 
-if __name__ == '__main__':
+def main():
     module = ForemanEntityAnsibleModule(
         argument_spec=dict(
             name=dict(required=True),
             updated_name=dict(),
+            compute_attributes=dict(type='list'),
         ),
         supports_check_mode=True,
     )
-    changed = main(module)
+
+    (compute_profile_dict, state) = module.parse_params()
+    name = compute_profile_dict.get('name')
+    updated_name = compute_profile_dict.get('updated_name')
+    compute_attributes = compute_profile_dict.pop('compute_attributes', list())
+
+    module.connect()
+
+    try:
+        # Try to find the compute_profile to work on
+        compute_profile = find_compute_profile(module, name=compute_profile_dict['name'], failsafe=True)
+    except Exception as e:
+        module.fail_json(msg='Failed to find entity: %s ' % e)
```

Comment:
Again: `Failsafe=False` ?

Reply:
Again :)

Reply:
I am not strongly against `failsafe=False` but leaving it would emphasize the `True` case and be consistent with the other modules.